platforms/enterprise/enterprise/src/main/java/org/gradle/internal/enterprise/impl/DefaultDevelocityBuildLifecycleService.java

```diff
@@ -19,19 +19,39 @@
 
 import org.gradle.api.Action;
 import org.gradle.api.Project;
+import org.gradle.api.configuration.BuildFeatures;
 import org.gradle.api.invocation.Gradle;
 import org.gradle.internal.enterprise.DevelocityBuildLifecycleService;
 
+import javax.inject.Inject;
+
 public class DefaultDevelocityBuildLifecycleService implements DevelocityBuildLifecycleService {
 
     private final Gradle gradle;
+    private final BuildFeatures features;
 
-    public DefaultDevelocityBuildLifecycleService(Gradle gradle) {
+    @Inject
+    public DefaultDevelocityBuildLifecycleService(Gradle gradle, BuildFeatures features) {
         this.gradle = gradle;
+        this.features = features;
     }
 
     @Override
     public void beforeProject(Action<? super Project> action) {
-        gradle.getLifecycle().beforeProject(action::execute);
+        // Preserve behavior when Isolated Projects is not enabled:
+        // - `allprojects` executes eagerly before any project has been evaluated, allowing its effects
+        //   to be observable from other eager configuration blocks (e.g., `subprojects { ... }`).
+        // - `lifecycle.beforeProject` executes just before each project is evaluated. Therefore, its effects
+        //   are not observable from eager configuration blocks, which would anyway be incompatible with
+        //   Isolated Projects.
+        if (isIsolatedProjects()) {
+            gradle.getLifecycle().beforeProject(action::execute);
+        } else {
+            gradle.allprojects(action);
+        }
```

Comment:
🤔 That is not great, given that we exercise the new code path only when isolated projects is on. I suppose there wasn't a way to make this work for both cases easily. Do you have a plan to do some changes in 8.9 so we also go through `lifecycle.beforeProject()`?

Reply:
We are currently working on running `isolatedProjectsIntegTest` on CI which will exercise this code path.

Reply:
See 050e5b41489b27e072873946d2cd34918bdf345c